pkg/controller/migration/convert/network.go

```diff
@@ -0,0 +1,165 @@
+	netBackend, err := c.node.getEnv(ctx, c.client, containerCalicoNode, "CALICO_NETWORKING_BACKEND")
+	if err != nil {
+		return err
+	}
+	if netBackend != nil && *netBackend != "" && *netBackend != "bird" {
+		return ErrIncompatibleCluster{"only CALICO_NETWORKING_BACKEND=bird is supported at this time"}
+	}
+
+	// Calico CNI
+	if c.calicoCNIConfig == nil {
+		return fmt.Errorf("no 'calico' cni conf found in CNI_NETWORK_CONFIG on install-cni")
+	}
+
+	// FELIX_DEFAULTENDPOINTTOHOSTACTION
+	defaultWepAction, err := c.node.getEnv(ctx, c.client, containerCalicoNode, "FELIX_DEFAULTENDPOINTTOHOSTACTION")
+	if err != nil {
+		return err
+	}
+	if defaultWepAction != nil && strings.ToLower(*defaultWepAction) != "accept" {
+		return ErrIncompatibleCluster{
+			fmt.Sprintf("unexpected FELIX_DEFAULTENDPOINTTOHOSTACTION: '%s'. Only 'accept' is supported.", *defaultWepAction),
+		}
+	}
+
+	// IP
+	ipMethod, err := c.node.getEnv(ctx, c.client, containerCalicoNode, "IP")
+	if err != nil {
+		return err
+	}
+	if ipMethod != nil && strings.ToLower(*ipMethod) != "autodetect" {
+		return ErrIncompatibleCluster{
+			fmt.Sprintf("unexpected IP value: '%s'. Only 'autodetect' is supported.", *ipMethod),
+		}
+	}
+
+	// IP_AUTODETECTION_METHOD
+	am, err := c.node.getEnv(ctx, c.client, containerCalicoNode, "IP_AUTODETECTION_METHOD")
+	if err != nil {
+		return err
+	}
+	if am != nil {
+		tam, err := getAutoDetection(*am)
+		if err != nil {
+			return err
+		}
+		install.Spec.CalicoNetwork.NodeAddressAutodetectionV4 = &tam
+	}
+
+	// CNI portmap plugin
+	if _, ok := c.pluginCNIConfig["portmap"]; ok {
+		hp := v1.HostPortsEnabled
+		install.Spec.CalicoNetwork.HostPorts = &hp
+	} else {
+		hp := v1.HostPortsDisabled
+		install.Spec.CalicoNetwork.HostPorts = &hp
+	}
+
+	// CNI bandwidth plugin
+	if _, ok := c.pluginCNIConfig["bandwidth"]; ok {
+		return ErrIncompatibleCluster{"operator does not yet support bandwidth"}
+	}
+
+	if c.calicoCNIConfig == nil {
+		// TODO: don't return an error once we support this, instead just returning nil.
+		return ErrIncompatibleCluster{"operator does not yet support running without calico CNI"}
+	}
+
+	if c.calicoCNIConfig.MTU == -1 {
+		// if MTU is -1, we assume it was us who replaced it when doing initial CNI
+		// config loading. We need to pull it from the correct source
+		mtu, err := c.node.getEnv(ctx, c.client, containerInstallCNI, "CNI_MTU")
+		if err != nil {
+			return err
+		}
+		if mtu == nil {
+			return fmt.Errorf("couldn't detect MTU")
+		}
+
+		// TODO: clean this up
+		i := intstr.FromString(*mtu)
+		iv := int32(i.IntValue())
+		install.Spec.CalicoNetwork.MTU = &iv
+	} else {
+		// user must have hardcoded their CNI instead of using our cni templating engine.
+		// use the hardcoded value.
+		mtu := int32(c.calicoCNIConfig.MTU)
+		install.Spec.CalicoNetwork.MTU = &mtu
+	}
+
+	// CNI IPAM pools
+	if len(c.calicoCNIConfig.IPAM.IPv4Pools) != 0 {
+		return ErrIncompatibleCluster{"cni ipam ranges not suported"}
```

Comment:
I think this should be `cni config assigning specified IPPools is not supported`. WDYT?

Reply:
I went with `specification of IPPools via cni config is not supported.` WDTY?

Reply:
I'm just trying to think what would be the most understandable for a user. WDYT about including `ipv4_pool` in the message. Maybe we could just say "specifying ipv4_pools in cni config is not supported"?